### ISSUE TITLE
Add HTTP identifier headers

### ISSUE THREAD

**Issue by bbockelm** [opensciencegrid/StashCache#22]
`stashcp` should populate the user-agent field so we can distinguish it at the various web services (such as the GeoIP application).

We should coordinate with Dave Dykstra to see what the best user identifier exists.

**Comment by djw8605** [comment]
Email to Dave Dykstra sent.

**Comment by bbockelm** [comment]
From the email thread:
* We should set the `User-Agent` header.  That's what CVMFS utilizes.
* Identify ourselves as `stashcp` and a particular version number.
* Additionally, we can utilize `X-Frontier-Id`.  Include the user identity (parse from the `$_CONDOR_JOB_AD`?) and the condor global job ID.

For example, here's the existing header for a CMS job generated by the frontier client:

```
wmagent_prozober_recovery-2-fabozzi_Run2016H-v1-ZeroBiasIsolatedBunch5-09Nov2016_8023__161201_144853_7563 2.8.19 2683277 cmsprod(9811) /DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=username/CN=718748/CN=MY USER NAME/CN=450678657/CN=631063546/CN=2596 -
```

We should use the same header name (otherwise all squids in WLCG need to be updated) but can make up our own format for the value.

**Comment by djw8605** [comment]
By same header name, do you mean the `X-Frontier-Id`?  `User-Agent` is standard, right?

**Comment by DrDaveD** [comment]
The first field in X-Frontier-ID comes from $FRONTIER_ID if it is set, otherwise $CMSSW_VERSION.

The other option is to use the Cvmfs-info header, that is also logged by frontier-squid.

**Comment by bbockelm** [comment]
Yeah - I'd like to keep the usage of `User-Agent` fairly standard (I don't want lots of unique strings here).  So, for the things that vary (username, job ID), I want to pick either `X-Frontier-ID` or `Cvmfs-info`.  I think `X-Frontier-ID` is a better name in the end...
